content/en/blog/_posts/2022-04-15-grpc-probes-in-beta.md

```diff
@@ -0,0 +1,208 @@
+basic search on GitHub (at the moment of writing). This is good indication of the tool popularity
+and the need to support this natively.
+
+Kubernetes v1.23 introduced an alpha-quality implementation of native support for
+querying a workload status using gRPC. Because it was an alpha feature,
+this was disabled by default for the v1.23 release.
+
+## Using the feature
+
+We built gRPC health checking in similar way with other probes and believe
+it will be [easy to use](/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes/#define-a-grpc-liveness-probe)
+if you are familiar with other probe types in Kubernetes.
+The natively supported health probe has many benefits over the workaround involving `grpc_health_probe` executable.
+
+With the native gRPC support you don't need to download and carry `10MB` of an additional executable with your image.
+Exec probes are generally slower than a gRPC call as they require instantiating a new process to run an executable.
+It also makes the checks less sensible for edge cases when the pod is running at maximum resources and has troubles
+instantiating new processes.
+
+There are a few limitations though. Since configuring a client certificate for probes is hard,
+services that require client authentication are not supported. The built-in probes are also
+not checking the server certificates and ignore related problems.
+
+Built-in checks also cannot be configured to ignore certain types of errors
+(`grpc_health_probe` returns different exit codes for different errors),
+and cannot be "chained" to run the health check on multiple services in a single probe.
+
+But all these limitations are quite standard for gRPC and there are easy workarounds
+for those.
+
+## Try it for yourself
+
+### Getting cluster
+
+You can try this feature today. To try the feature you can spin up the kubernetes cluster
+yourself with the `GRPCContainerProbe` feature gate enabled, there are many [tools available](/docs/tasks/tools/).
+
+Since the feature gate `GRPCContainerProbe` is enabled by default in 1.24,
+many vendors will have this functionality working out of the box.
+So you may just create an 1.24 cluster on platform of your choice. Some vendors
+allow to enable alpha features on 1.23 clusters.
+
+For example, at the moment of writing, you can spin up the test cluster on GKE for a quick test.
+Other vendors may also have similar capabilities, especially if you
+read this blog post long after kubernetes 1.24 released.
+
+On GKE use the following command (note, version is `1.23` and `enable-kubernetes-alpha` are specified).
+
+```shell
+gcloud container clusters create test-grpc \
+    --enable-kubernetes-alpha \
+    --no-enable-autorepair \
+    --no-enable-autoupgrade \
+    --release-channel=rapid \
+    --cluster-version=1.23
+```
+
+You will also need to configure `kubectl` to access the cluster:
+
+```shell
+gcloud container clusters get-credentials test-grpc
+```
```

Comment:
I'd definitely prefer an example that wasn't cloud-provider-specific. Nudging readers towards GKE isn't the right look.

It's OK to provide a _set_ of examples; for example, you could have examples for:
- GKE
- kind
- minikube

(we prefer to list in alphabetical order)

Reply:
nudging wasn't an intention, I just wanted to provide something easier than kind to make people try it. Kind instructions will be a bit more involved that I'd prefer - changing feature gates will be needed.

if this is not ideal, I can remove that part completely only leaving the references to `kind`. Not sure if providing step-by-step instruction would be useful for anybody. People who will try it with kind, likely already know it. People who doesn't - don't need those.

wdyt?

Reply:
I'm fine with either option.